### DISCUSSION TITLE
Possibly-undefined store

### DISCUSSION THREAD

**Discussion by tpict** [pmndrs/zustand#669] [Q&A] [answered]
Trying to do something like

```tsx
import create from "zustand";
const useStore = create<string | undefined>(undefined);
```

results in a TS `Type 'string | undefined' does not satisfy the constraint 'object'.` error.

Is this a deliberate constraint, i.e. Zustand doesn't support transitioning a state to certain primitives?

**Comment by dai-shi** [comment, marked as answer]
Yes, it has to be an object.
